src/batched/sparse/impl/KokkosBatched_GMRES_TeamVector_Impl.hpp

```diff
@@ -181,9 +185,12 @@ struct TeamVectorGMRES {
             Kokkos::TeamVectorRange(member, 0, numMatrices),
             [&](const OrdinalType& ii) { tmp(ii) = -tmp(ii); });
 
+        member.team_barrier();  // Finish writing to tmp
+
         TeamVectorAxpy<MemberType>::invoke(member, tmp, V_i, W);
       }
 
+      member.team_barrier();  // Finish writing to W
```

Comment:
I am not sure if it is needed but I think it is safer to move this team barrier just after the `TeamVectorAxpy<MemberType>::invoke(member, tmp, V_i, W);` within the `for (size_t i = 0; i < j + 1; ++i) {}` loop.